api/v1/services/google_oauth.py

```diff
@@ -0,0 +1,197 @@
+from fastapi import Depends, HTTPException, status
+from datetime import datetime, timezone
+from api.db.database import get_db
+from api.v1.models.user import User
+from api.v1.models.oauth import OAuth
+from api.v1.models.profile import Profile
+from api.core.base.services import Service
+from sqlalchemy.orm import Session
+from typing import Annotated, Union
+from api.v1.services.user import user_service
+from api.v1.schemas.google_oauth import Tokens, UserData, StatusResponse
+
+
+class GoogleOauthServices(Service):
+    """
+    Handles database operations for google oauth
+    """
+    def create(self, google_response: dict,
+               db: Annotated[Session, Depends(get_db)]) -> object:
+        """
+        Creates a user using information from google.
+
+        Args:
+            google_response: Raw data from google oauth2
+            db: database session to manage database operation
+
+        Returns:
+            user: The user object if already exists or if newly created
+            Response: HttpException for when Authentication fails
+        """
+        try:
+            # retrieve the user information
+            user_info: dict = google_response.get("userinfo")
+
+            existing_user = db.query(User).filter_by(email=user_info.get("email")).one_or_none()
+
+            if existing_user:
+                # retrieve the user's google_access_token
+                oauth_data = db.query(OAuth).filter_by(user_id=existing_user.id).one_or_none()
+                # if the entry exists
+                if oauth_data:
+                    # update the oauth data
+                    self.update(oauth_data, google_response, db)
+                    # pass the user object to get_response method to generate a response object
+                    user_response = self.get_response(existing_user)
+                    return user_response
```

Comment:
moving forward, can this all process be streamlined and handled more effectively